src/Components/Endpoints/src/RazorComponentEndpointInvoker.cs

```diff
@@ -165,7 +164,9 @@ private async Task<RequestValidationState> ValidateRequestAsync(HttpContext cont
 
         if (processPost)
         {
-            if (!context.Request.HasFormContentType)
```

Comment:
@captainsafia I think this was triggered by the antiforgery middleware, I believe we should revisit this to avoid throwing and instead return false. It's surprising otherwise and might break existing code, do you have thoughts?

Reply:
I think you might be talking about the exception that gets thrown [here](https://source.dot.net/#Microsoft.AspNetCore.Http/Features/FormFeature.cs,189)). If so, I'm a little cautious about changing this behavior given its in the FormFeature shared across the stack as opposed to just being in anti-forgery middleware.

We don't eagerly validate the content type in the anti-forgery middleware (although we could) then we wouldn't fall through to the exception in the FormFeature.

Reply:
I might have been confused and thought the issue might have been related to antiforgery, since I saw an exception pop right before.

I'll take another quick look at it and see if I repro it in a simpler way. I throught it was the thing where we poisoned the IFormFeature that was throwing, but I can't find it right now.